realm-transformer/src/main/groovy/io/realm/transformer/RealmOptionalAPITransformer.groovy

```diff
@@ -0,0 +1,183 @@
+package io.realm.transformer
+
+import com.android.SdkConstants
+import com.android.build.api.transform.Format
+import groovy.io.FileType
+import io.realm.annotations.internal.OptionalAPI
+
+import com.android.build.api.transform.Context
+import com.android.build.api.transform.QualifiedContent
+import com.android.build.api.transform.Transform
+import com.android.build.api.transform.TransformException
+import com.android.build.api.transform.TransformInput
+import com.android.build.api.transform.TransformOutputProvider
+import com.google.common.collect.ImmutableSet
+import com.google.common.collect.Sets
+import javassist.ClassPool
+import javassist.CtMethod
+import javassist.LoaderClassPath
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import java.util.jar.JarFile
+import java.util.regex.Pattern
+
+import static com.android.build.api.transform.QualifiedContent.*
+
+class RealmOptionalAPITransformer extends Transform {
+
+    private Logger logger = LoggerFactory.getLogger('realm-logger')
+
+    @Override
+    String getName() {
+        return "RealmOptionalAPITransformer"
+    }
+
+    @Override
+    Set<QualifiedContent.ContentType> getInputTypes() {
+        return ImmutableSet.<ContentType> of(DefaultContentType.CLASSES)
+    }
+
+    @Override
+    Set<QualifiedContent.Scope> getScopes() {
+        return Sets.immutableEnumSet(Scope.EXTERNAL_LIBRARIES)
+    }
+
+    @Override
+    Set<Scope> getReferencedScopes() {
+        return Sets.immutableEnumSet(Scope.PROJECT, Scope.PROJECT_LOCAL_DEPS, Scope.SUB_PROJECTS,
```

Comment:
That is the reference scope is used to identify if the rx.observable exists. So i think it should contain all of them?